### PR TITLE
chore(codebuild): aarch64 standard 2.0 and 3.0 build images are not deprecated

### PR DESCRIPTION
**Pull request by nikovirtala**
### Issue # (if applicable)

Closes #<issue number here>.

### Reason for this change

AWS CodeBuild Docker images `AMAZON_LINUX_2_ARM_3` aka `AMAZON_LINUX_2_STANDARD_3_0` aka `aws/codebuild/amazonlinux2-aarch64-standard:3.0` and `AMAZON_LINUX_2_ARM_2` aka `AMAZON_LINUX_2_STANDARD_2_0` aka `aws/codebuild/amazonlinux2-aarch64-standard:2.0`are not deprecated but the currently actively maintained ones according to https://github.com/aws/aws-codebuild-docker-images and https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-available.html

### Description of changes

I removed the JSDoc Deprecated flag for the currently active images and added their image names as a comment so that one can find the images also using those references.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by nikovirtala** [comment]
Exemption Request: this change doesn't change any feature or functionality — thus, no test should and does not change

File: packages/aws-cdk-lib/aws-codebuild/lib/project.ts [hunk 1]
```diff
@@ -1795,7 +1795,7 @@ export class LinuxBuildImage implements IBuildImage {
    * @deprecated Use LinuxArmBuildImage.AMAZON_LINUX_2_STANDARD_3_0 instead.
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-codebuild/lib/project.ts:1795]: v2 seems not to be deprecated as well.
Can you please update the comment accordingly?

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks 👍
Left a comment for updating v2 documentation as well.
I think you can mark the PR as a `chore` (CI won't require tests).
Also, can you please update the title to not be truncated?

**Comment by nikovirtala** [comment]
> Thanks 👍 Left a comment for updating v2 documentation as well. I think you can mark the PR as a `chore` (CI won't require tests). Also, can you please update the title to not be truncated?

I removed the `@deprecated` flag also for v2 image and update the pr title and description to cover that change too.

**Comment by nmussy** [comment]
Hey, guess I didn't open a PR for this (I was sure I did), but I already have a runner ready to update all of the CodeBuild images (`WindowsBuildImage`, `LinuxBuildImage`, `LinuxArmBuildImage`, `LinuxLambdaBuildImage`, and `LinuxArmLambdaBuildImage`). I'll try to take care of this soon.

Regardless, you can confirm that these images are still available by using the [`codebuild:ListCuratedEnvironmentImages`](https://docs.aws.amazon.com/codebuild/latest/APIReference/API_ListCuratedEnvironmentImages.html) API command:

```sh
$ aws codebuild list-curated-environment-images | jq '.platforms[] | select(.platform == "AMAZON_LINUX_2") | .languages[] | select(.language == "STANDARD") | .images[] | select(.name == "aws/codebuild/amazonlinux2-aarch64-standard:3.0")'

{
  "name": "aws/codebuild/amazonlinux2-aarch64-standard:3.0",
  "description": "AWS CodeBuild - Docker image based on Amazon Linux 2023 (aarch64) with multiple language support",
  "versions": [
    "aws/codebuild/amazonlinux2-aarch64-standard:3.0-23.04.25",
    "aws/codebuild/amazonlinux2-aarch64-standard:3.0-24.01.29",
    "aws/codebuild/amazonlinux2-aarch64-standard:3.0-24.03.06",
    "aws/codebuild/amazonlinux2-aarch64-standard:3.0-24.03.21"
  ]
}
```

```sh
$ aws codebuild list-curated-environment-images | jq '.platforms[] | select(.platform == "AMAZON_LINUX_2") | .languages[] | select(.language == "STANDARD") | .images[] | select(.name == "aws/codebuild/amazonlinux2-aarch64-standard:2.0")'
{
  "name": "aws/codebuild/amazonlinux2-aarch64-standard:2.0",
  "description": "AWS CodeBuild - Docker image based on Amazon Linux 2 (aarch64) with multiple language support",
  "versions": [
    "aws/codebuild/amazonlinux2-aarch64-standard:2.0-20.06.15",
    "aws/codebuild/amazonlinux2-aarch64-standard:2.0-20.08.13",
    "aws/codebuild/amazonlinux2-aarch64-standard:2.0-21.01.12",
    "aws/codebuild/amazonlinux2-aarch64-standard:2.0-21.03.02",
    "aws/codebuild/amazonlinux2-aarch64-standard:2.0-21.08.20",
    "aws/codebuild/amazonlinux2-aarch64-standard:2.0-21.10.15",
    "aws/codebuild/amazonlinux2-aarch64-standard:2.0-24.01.29",
    "aws/codebuild/amazonlinux2-aarch64-standard:2.0-24.03.06",
    "aws/codebuild/amazonlinux2-aarch64-standard:2.0-24.03.21"
  ]
}
```

**Comment by nmussy** [comment]
I opened another PR to fix this issue, among other things: #29616
Sorry about that, I had everything ready but forgot to actually make the changes 😞

**Review by TheRealAmazonKendra — APPROVED** [review]

**Comment by TheRealAmazonKendra** [comment]
@Merfigyio update

**Review by TheRealAmazonKendra — APPROVED** [review]

**Pull request merged**